biz.aQute.bndlib/src/aQute/bnd/classfile/ClassFile.java

```diff
@@ -108,6 +112,10 @@ public static ClassFile parseClassFile(DataInput in) throws IOException {
 		return class_file;
 	}
 
+	public static ClassFile parseInputStream(InputStream in) throws IOException {
+		return parseClassFile(new DataInputStream(in));
```

Comment:
I think you need to do something here to close the input stream. DataInput does not have a close so parseClassFile(DataInput) will not be able to close the input stream. 

Properly closing the input stream will be especially important for your new parseFile method so we don't lock files open on Windows.